tests/basic/async_dr_test.go

```diff
@@ -637,13 +639,274 @@ var _ = Describe("{StorkctlPerformFailoverFailbackeckEsClusterwide}", func() {
+		err = Inst().S.RefreshNodeRegistry()
+		log.FailOnError(err, "Node registry refresh failed")
+		err = Inst().V.RefreshDriverEndpoints()
+		log.FailOnError(err, "Refresh Driver end points failed")
+
+		Step("start the upgrade of volume driver", func() {
+			log.InfoD("start the upgrade of volume driver")
+
+			if len(Inst().UpgradeStorageDriverEndpointList) == 0 {
+				log.Fatalf("Unable to perform volume driver upgrade hops, none were given")
+			}
+
+			// Perform upgrade hops of volume driver based on a given list of upgradeEndpoints passed
+			for _, upgradeHop := range strings.Split(Inst().UpgradeStorageDriverEndpointList, ",") {
+				currPXVersion, err := Inst().V.GetDriverVersionOnNode(storageNodes[0])
+				if err != nil {
+					log.Warnf("error getting driver version, Err: %v", err)
+				}
+				upgradeStatus, updatedPXVersion, durationInMins := upgradePX(upgradeHop, storageNodes)
+				log.InfoD("current PX version: %s, no of nodes: %d, upgrade status: %s, updated px version: %s, duration in mins: %d on source cluster",
+					currPXVersion, numOfNodes, upgradeStatus, updatedPXVersion, durationInMins)
+				err = hardSetConfig(kubeConfigPathDest)
+				log.FailOnError(err, "Switching context to destination cluster failed")
+				err = SetCustomKubeConfig(asyncdr.SecondCluster)
+				log.FailOnError(err, "Switching context to destination cluster failed")
+				err = Inst().S.RefreshNodeRegistry()
+				log.FailOnError(err, "Node registry refresh failed")
+				err = Inst().V.RefreshDriverEndpoints()
+				log.FailOnError(err, "Refresh Driver end points failed")
+				currPXVersion, err = Inst().V.GetDriverVersionOnNode(storageNodesDest[0])
+				if err != nil {
+					log.Warnf("error getting driver version, Err: %v", err)
+				}
+				upgradeStatus, updatedPXVersion, durationInMins = upgradePX(upgradeHop, storageNodesDest)
+				log.InfoD("current PX version: %s, no of nodes: %d, upgrade status: %s, updated px version: %s, duration in mins: %d, on dest cluster",
+					currPXVersion, numOfNodesDest, upgradeStatus, updatedPXVersion, durationInMins)
+				err = hardSetConfig(kubeConfigPathSrc)
+				log.FailOnError(err, "Switching context to source cluster failed")
+				err = SetCustomKubeConfig(asyncdr.FirstCluster)
+				log.FailOnError(err, "Switching context to source cluster failed")
+				err = Inst().S.RefreshNodeRegistry()
+				log.FailOnError(err, "Node registry refresh failed")
+				err = Inst().V.RefreshDriverEndpoints()
+				log.FailOnError(err, "Refresh Driver end points failed")
+				majorVersion := strings.Split(currPXVersion, "-")[0]
+				statsData := make(map[string]string)
+				statsData["numOfNodes"] = fmt.Sprintf("%d", numOfNodes)
+				statsData["fromVersion"] = currPXVersion
+				statsData["toVersion"] = updatedPXVersion
+				statsData["duration"] = fmt.Sprintf("%d mins", durationInMins)
+				statsData["status"] = upgradeStatus
+				dash.UpdateStats("px-upgrade-stats", "px-enterprise", "upgrade", majorVersion, statsData)
```

Comment:
Can we have something like?

```go
clusters := []struct {
    kubeConfigPath string
    clusterName    string
    storageNodes   []string
}{
    {kubeConfigPathDest, asyncdr.SecondCluster, storageNodesDest},
    {kubeConfigPathSrc, asyncdr.FirstCluster, storageNodes},
}

for _, cluster := range clusters {
    err := hardSetConfig(cluster.kubeConfigPath)
    log.FailOnError(err, fmt.Sprintf("Switching context to cluster with kubeConfigPath: %s failed", cluster.kubeConfigPath))
    err = SetCustomKubeConfig(cluster.clusterName)
    log.FailOnError(err, fmt.Sprintf("Switching context to cluster with kubeConfigPath: %s failed", cluster.kubeConfigPath))
    err = Inst().S.RefreshNodeRegistry()
    log.FailOnError(err, fmt.Sprintf("Node registry refresh failed for cluster with kubeConfigPath: %s", cluster.kubeConfigPath))
    err = Inst().V.RefreshDriverEndpoints()
    log.FailOnError(err, fmt.Sprintf("Refresh Driver end points failed for cluster with kubeConfigPath: %s", cluster.kubeConfigPath))

    currPXVersion, err := Inst().V.GetDriverVersionOnNode(cluster.storageNodes[0])
    if err != nil {
        log.Warnf("error getting driver version for cluster with kubeConfigPath: %s, Err: %v", cluster.kubeConfigPath, err)
    }

    upgradeStatus, updatedPXVersion, durationInMins := upgradePX(upgradeHop, cluster.storageNodes)

    log.InfoD("current PX version: %s, no of nodes: %d, upgrade status: %s, updated px version: %s, duration in mins: %d on cluster with kubeConfigPath: %s",
        currPXVersion, len(cluster.storageNodes), upgradeStatus, updatedPXVersion, durationInMins, cluster.kubeConfigPath)

    majorVersion := strings.Split(currPXVersion, "-")[0]
    statsData := make(map[string]string)
    statsData["numOfNodes"] = fmt.Sprintf("%d", len(cluster.storageNodes))
    statsData["fromVersion"] = currPXVersion
    statsData["toVersion"] = updatedPXVersion
    statsData["duration"] = fmt.Sprintf("%d mins", durationInMins)
    statsData["status"] = upgradeStatus

    dash.UpdateStats("px-upgrade-stats", "px-enterprise", "upgrade", majorVersion, statsData)
}
```

Reply:
Thank you, just did changes with your help.